osidb/sync_manager.py

```diff
@@ -0,0 +1,614 @@
+            # 2) Failed recently
+            # 3) Not permanent failure
+            # 3) Failed more than FAIL_RESCHEDULE_DELAY ago
+            # 4) Was not scheduled after last failure
+            #
+            #                 |     FAIL_RESCHEDULE_DELAY    |
+            #     |-----------|------------------------------|---//----------?
+            #  Started      Fail (not permanent?)           NOW          Scheduled
+            #
+
+            if (
+                sync_manager.last_started_dt is not None
+                and 0 < sync_manager.last_consecutive_failures
+                and not sync_manager.permanently_failed
+                and timezone.now() - sync_manager.last_failed_dt
+                > cls.FAIL_RESCHEDULE_DELAY
+                and sync_manager.last_scheduled_dt < sync_manager.last_failed_dt
+            ):
+                sync_manager.reschedule(
+                    f"Failed {sync_manager.last_consecutive_failures} times"
+                )
+                continue
+
+    def __str__(self):
+        self.refresh_from_db()
+        result = ""
+        result += f"Sync ID: {self.sync_id}\n"
+        result += f"Scheduled: {self.last_scheduled_dt}\n"
+        result += f"Started: {self.last_started_dt}\n"
+        result += f"Finished: {self.last_finished_dt}\n"
+        result += f"Failed: {self.last_failed_dt}\n"
+        result += f"Failed consecutive: {self.last_consecutive_failures}\n"
+        result += f"Failed reason: {self.last_failed_reason}\n"
+        result += f"Rescheduled: {self.last_rescheduled_dt}\n"
+        result += f"Rescheduled consecutive: {self.last_consecutive_reschedules}\n"
+        result += f"Rescheduled reason: {self.last_rescheduled_reason}\n"
+        return result
+
+
+class FlawDownloadManager(SyncManager):
+    """
+    Sync manager class for Bugzilla => OSIDB Flaw synchronization.
+    """
+
+    @staticmethod
+    @app.task(name="sync_manager.flaw_download", bind=True)
+    def sync_task(self, flaw_id):
+        from collectors.bzimport import collectors
+
+        download = FlawDownloadManager.get_sync_manager(flaw_id)
+        download.started(self)
+
+        set_user_acls(settings.ALL_GROUPS)
+
+        # Code adapted from collectors.bzimport.collectors.FlawCollector.collect
+        collector = collectors.FlawCollector()
+        try:
+            collector.sync_flaw(flaw_id)
+        except Exception as e:
+            download.failed(e)
+        else:
+            download.finished()
+        finally:
+            collector.free_queriers()
+
+    def update_synced_links(self):
+        from osidb.models import Flaw
+
+        flaws = Flaw.objects.filter(meta_attr__bz_id=self.sync_id)
+        flaws.update(download_manager=self)
+
+    def __str__(self):
+        from osidb.models import Flaw
+
+        result = super().__str__()
+
+        flaws = Flaw.objects.filter(meta_attr__bz_id=self.sync_id)
+        cves = [f.cve_id for f in flaws]
```

Comment:
I would suggest to use IDs instead of CVEs in form
```
f.cve_id or f.uuid
```
or you can end up with `None` values.

Reply:
Actually BZ ID is I think what we want so we can identify the entity to be synced from BZ.

Reply:
BZ ID is printed in super.